tools/protoc-gen-chpl/Makefile

```diff
@@ -0,0 +1,62 @@
+# Copyright 2020 Hewlett Packard Enterprise Development LP
+# Copyright 2004-2019 Cray Inc.
+# Other additional copyright holders may be indicated within.
+#
+# The entirety of this work is licensed under the Apache License,
+# Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License.
+#
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+ifndef CHPL_MAKE_HOME
+export CHPL_MAKE_HOME=$(realpath $(shell pwd)/../..)
+endif
+
+include $(CHPL_MAKE_HOME)/make/Makefile.base
+
+unexport CHPL_MAKE_CHPLENV_CACHE
+
+bdir=$(CHPL_BIN_DIR)
+link=../../tools/protoc-gen-chpl/protoc-gen-chpl
+linkFile=$(bdir)/protoc-gen-chpl
+
+SOURCES=$(wildcard *.cpp)
+OBJ=$(SOURCES:.cpp=.o)
+LDFLAGS = -lprotobuf -lprotoc
+CPPFLAGS = -I.
+
+all: protoc-gen-chpl
+
+protoc-gen-chpl:	$(OBJ)
+	@echo "Building protoc-gen-chpl.."
+	$(CC) -o $@ $^ $(LDFLAGS) -lstdc++
+
+install: protoc-gen-chpl
+ifneq ($(wildcard $(linkFile)),)
```

Comment:
I don't really understand the purpose of the symbolic link business but we were doing it for mason and this code can follow that for now.

Reply:
It seems fine to me, it is just creating a symbolic link `/chapel/bin/linux64-x86_64/protoc-gen-chpl` (in Linux 64 bit case) and `tools/protoc-gen-chpl/protoc-gen-chpl`. This is necessary because only `/chapel/bin` is accessible through the system wide `PATH` variable. If that makes sense?

Reply:
Either way. I expect to change it later (along with mason) to not use this approach. It should just put the binary in the bin directory and not also in the tools directory.

Reply:
> . It should just put the binary in the bin directory and not also in the tools directory.

Yes, that sounds good.

Reply:
See also PR #16154.